### PR TITLE
refactor center onMessage, and create UT

### PR DESCRIPTION
**Pull request by hydraxman**
**Linked GitHub issue ID**: #  

## Pull Request Checklist

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [x] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [x] No

## Pull Request Description

**A few words to explain your changes**:  refactor center onMessage, and create UT

### How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Build related changes
- [x] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*

### CODE REVIEW

File: center/src/test/java/com/microsoft/hydralab/center/service/DeviceAgentManagementServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @Resource
+    DeviceAgentManagementService deviceAgentManagementService;
+    @MockBean
+    AgentUserRepository agentUserRepository;
+    @Resource
+    ApplicationContext applicationContext;
+
+
+    @Test
+    public void testOnMessage() throws IOException {
```
Comment (zhou9584) [inline center/src/test/java/com/microsoft/hydralab/center/service/DeviceAgentManagementServiceTest.java:34]: This case failed in the pipeline. Please help to check.

File: center/src/test/java/com/microsoft/hydralab/center/service/DeviceAgentManagementServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        StorageServiceClientProxy storageServiceClientProxy = new StorageServiceClientProxy(applicationContext);
+        storageServiceClientProxy.initCenterStorageClient("LOCAL");
+        given(this.storageServiceClientProxy.generateAccessToken(Const.FilePermission.WRITE)).willReturn(storageServiceClientProxy.generateAccessToken(Const.FilePermission.WRITE));
+
+        Message message = Message.ok(Const.Path.AUTH, agentUser);
+        byte[] byteMsg = SerializeUtil.messageToByteArr(message);
+
+        deviceAgentManagementService.onMessage(SerializeUtil.byteArrToMessage(byteMsg), session);
+        session.close();
+    }
```
Comment (zhou9584) [inline center/src/test/java/com/microsoft/hydralab/center/service/DeviceAgentManagementServiceTest.java:64]:
```suggestion
    @Resource
    LocalStorageProperty localStorageProperty;
    @Resource
    AgentUserRepository agentUserRepository;
    @Resource
    DeviceAgentManagementService deviceAgentManagementService;

    @Test
    public void testOnMessage_NoException() throws IOException {
        Session session = Mockito.mock(Session.class);
        RemoteEndpoint.Basic basicRemote = Mockito.mock(RemoteEndpoint.Basic.class);
        Mockito.when(session.getId()).thenReturn("123456");
        Mockito.when(session.getBasicRemote()).thenReturn(basicRemote);

        String agentId = "agentId";
        AgentUser agentUser = new AgentUser();
        agentUser.setId(agentId);
        agentUser.setName("agentName");
        agentUser.setSecret("agentSecret");
        agentUser.setTeamName("Microsoft");
        InetAddress localHost = InetAddress.getLocalHost();
        agentUser.setHostname(localHost.getHostName());
        agentUser.setIp("127.0.0.1");
        agentUser.setOs(System.getProperties().getProperty("os.name"));
        agentUser.setVersionName("versionName");
        agentUser.setVersionCode("versionCode");

        agentUserRepository.saveAndFlush(agentUser);

        LocalStorageClientAdapter localStorageClientAdapter = new LocalStorageClientAdapter(localStorageProperty);
        given(this.storageServiceClientProxy.generateAccessToken(Const.FilePermission.WRITE)).willReturn(localStorageClientAdapter.generateAccessToken(Const.FilePermission.WRITE));

        Message message = Message.ok(Const.Path.AUTH, agentUser);
        byte[] byteMsg = SerializeUtil.messageToByteArr(message);

        deviceAgentManagementService.onMessage(SerializeUtil.byteArrToMessage(byteMsg), session);
        session.close();
    }
```

**Comment by hydraxman** [comment]
@zhou9584 could you help review this PR when it's a convenient time for you? Thanks!

And thank you so much for helping resolve the UT issue.

File: sdk/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# used for azure tts generation in a detached branch.
+scripts
```
Comment (olivershen-wow) [inline sdk/.gitignore:2]: What's this script folder for?

**Comment by olivershen-wow** [comment]
LGTM for the rest.

**Review by olivershen-wow — DISMISSED** [review]

File: sdk/.gitignore (hunk 3 above)
Comment (hydraxman) [reply sdk/.gitignore:2]: Should be not needed, I used it for a tts generate audio file storage, will remove this change, thanks.

Comment (hydraxman) [reply sdk/.gitignore]: @olivershen-wow removed the file, please help review again thanks.

**Review by olivershen-wow — APPROVED** [review]

**Pull request merged**